### PR TITLE
DRV-598: Add missing functions to C# driver

### PR DESCRIPTION
**Pull request by parkhomenko**
### Notes
Add new functions:
All
Any
IsArray
IsBoolean
IsBytes
IsCollection
IsCredentials
IsDatabase
IsDate
IsDoc
IsDouble
IsFunction
IsIndex
IsInteger
IsKey
IsLambda
IsNull
IsNumber
IsObject
IsRef
IsRole
IsSet
IsString
IsTimestamp
IsToken
TimeAdd
TimeSubtract
TimeDiff
ToArray
ToDouble
ToInteger
ToObject

### Jira Ticket
https://faunadb.atlassian.net/browse/DRV-598

### CODE REVIEW

**Review by vadimLF — DISMISSED** [review]

File: FaunaDB.Client.Test/ClientTest.cs [hunk 1]
```diff
@@ -2355,6 +2355,293 @@ public async Task TestHasCurrentTokenWithInternalKey()
...
+        }
+
+        [Test]
+        public async Task TestThrowBadRequestOnDouble()
+        {
+            var ex = Assert.ThrowsAsync<BadRequest>(
+                async () => await adminClient.Query(ToDouble(Now()))
+            );
+            Assert.AreEqual("invalid argument: Cannot cast Time to Double.", ex.Message);
+        }
```
Comment (evbruno) [inline FaunaDB.Client.Test/ClientTest.cs:2585]: I did appreciate this "error" tests!!! Thanks for adding it :D

File: FaunaDB.Client/Query/Language.Miscellaneous.cs [hunk 2]
```diff
@@ -920,5 +920,121 @@ public partial struct Language
...
+        
+        /// <summary>
+        /// Returns a new time or date with the offset in terms of the unit added.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/timeadd">TimeAdd</see>
+        /// </para>
+        /// </summary>
+        public static Expr TimeAdd(Expr value, Expr offset, FaunaTimeUnit unit) =>
+            TimeAdd(value, offset, unit.Value);
+
```
Comment (evbruno) [inline FaunaDB.Client/Query/Language.Miscellaneous.cs:977]: I think it's  nice to wrap unit as a new type `FaunaTimeUnit`, but, can't we use an `enumerate` here ?

File: FaunaDB.Client/Query/Language.Miscellaneous.cs [hunk 3]
```diff
@@ -920,5 +920,121 @@ public partial struct Language
...
+        /// <summary>
+        /// Returns a new time or date with the offset in terms of the unit added.
+        /// <para>
+        /// See the <see href="https://docs.fauna.com/fauna/current/api/fql/functions/timeadd">TimeAdd</see>
+        /// </para>
+        /// </summary>
+        public static Expr TimeAdd(Expr value, Expr offset, Expr unit) =>
+            UnescapedObject.With("time_add", value, "offset", offset, "unit", unit);
+
```
Comment (evbruno) [inline FaunaDB.Client/Query/Language.Miscellaneous.cs:968]: WDYT about exploring `unit` as an explicit `String` here?

File: FaunaDB.Client/Query/Language.Miscellaneous.cs (hunk 2 above)
Comment (parkhomenko) [reply FaunaDB.Client/Query/Language.Miscellaneous.cs]: thanks a lot for noticing this one :)
I did a bad thing by introducing that new type.

File: FaunaDB.Client/Query/Language.Miscellaneous.cs (hunk 3 above)
Comment (parkhomenko) [reply FaunaDB.Client/Query/Language.Miscellaneous.cs]: sure, I've refactored the by adding the corresponding implicit and explicit conversions

**Review by evbruno — APPROVED** [review]
🚀

**Review by vadimLF — APPROVED** [review]
LGTM.

**Pull request merged**